Source/WebCore/platform/graphics/SourceBufferPrivate.h

```diff
@@ -74,8 +76,7 @@ enum class SourceBufferAppendMode : uint8_t {
 };
 
 class SourceBufferPrivate
-    : public RefCounted<SourceBufferPrivate>
-    , public CanMakeWeakPtr<SourceBufferPrivate>
+    : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<SourceBufferPrivate>
```

Comment:
It seems we need SourceBufferPrivate to being destroyed in main thread.
Maybe we should WTF::DestructionThread::MainThread for now?

Reply:
why would it need to be destroyed on the main thread?